### PR TITLE
Remove all usage of getSCMigration ref

### PR DESCRIPTION
**Pull request by craigpalermo**
This PR also removes usage of `mapProps` where possible in favor of SC attrs

- [x] fix snapshot failure in `Popover`
- [x] update snapshot for `Modal`

### CODE REVIEW

File: packages/core/src/BlockLink/BlockLink.js [hunk 1]
```diff
@@ -1,21 +1,13 @@
...
+const BlockLink = styled(Link)`
   display: block;
   color: inherit;
   text-decoration: none;
-`)
+`
 
 BlockLink.displayName = 'BlockLink'
 
-BlockLink.propTypes = {
```
Comment (sdalonzo) [inline packages/core/src/BlockLink/BlockLink.js:17]: should this use `Link.propTypes`?

File: packages/core/src/BlockLink/__snapshots__/BlockLink.spec.js.snap [hunk 2]
```diff
@@ -2,14 +2,26 @@
 
 exports[`BlockLink renders 1`] = `
 .c0 {
+  cursor: pointer;
+  -webkit-text-decoration: none;
+  text-decoration: none;
+  color: #007aff;
```
Comment (sdalonzo) [inline packages/core/src/BlockLink/__snapshots__/BlockLink.spec.js.snap:8]: 😱

File: packages/core/src/CloseButton/__snapshots__/CloseButton.spec.js.snap [hunk 3]
```diff
@@ -27,11 +27,18 @@ exports[`CloseButton renders without props 1`] = `
   border-style: solid;
   font-size: 14px;
   padding: 9.5px 18px;
+  background-color: #007aff;
```
Comment (sdalonzo) [inline packages/core/src/CloseButton/__snapshots__/CloseButton.spec.js.snap:30]: 🤔

File: packages/core/src/CloseButton/__snapshots__/CloseButton.spec.js.snap [hunk 4]
```diff
@@ -55,7 +62,9 @@ exports[`CloseButton renders without props 1`] = `
 }
 
 <button
+  aria-label="close"
```
Comment (sdalonzo) [inline packages/core/src/CloseButton/__snapshots__/CloseButton.spec.js.snap:65]: 😱

Comment (craigpalermo) [reply packages/core/src/CloseButton/__snapshots__/CloseButton.spec.js.snap:65]: Yeah, I have a couple issues to iron out haha

Comment (sdalonzo) [reply packages/core/src/CloseButton/__snapshots__/CloseButton.spec.js.snap:65]: this is good. it's bad that it was previously getting suppressed on derivative buttons.

File: packages/core/src/Link/Link.stories.js [hunk 5]
```diff
@@ -40,7 +40,7 @@ storiesOf('Link', module)
     <ForwardRefDemo
       refChild={(dsRef) => (
         <>
-          <Link color='darkGray' dsRef={dsRef}>
+          <Link color='darkGray' ref={dsRef}>
```
Comment (pcln-james) [inline packages/core/src/Link/Link.stories.js:43]: nb: opportunity to update this color to `text.dark`

**Review by pcln-james — APPROVED** [review]

**Review by pcln-barry — APPROVED** [review]
Looks good to me 👍

File: packages/core/src/BlockLink/BlockLink.stories.js [hunk 6]
```diff
@@ -25,7 +25,7 @@ storiesOf('BlockLink', module)
     <ForwardRefDemo
       refChild={(dsRef) => (
         <>
-          <BlockLink color='darkGray' dsRef={dsRef}>
+          <BlockLink color='darkGray' ref={dsRef}>
```
Comment (bmansell) [inline packages/core/src/BlockLink/BlockLink.stories.js:28]: Could we swap with `text.dark` while we're in here?

**Review by bmansell — APPROVED** [review]

**Pull request merged**